File: templates/service-java/{{#server.micronaut}}/src/main/java/{{package.value}}/{{#jsonlib.moshi}}/MoshiConfig.java
package {{package.value}};

import com.squareup.moshi.Moshi;
import io.micronaut.context.annotation.*;
import {{package.value}}.json.*;

@Factory
public class MoshiConfig {
    @Bean
    public Moshi getMoshi() {
        var moshiBuilder = new Moshi.Builder();
		CustomMoshiAdapters.setup(moshiBuilder);
        return moshiBuilder.build();
    }

    @Bean
    public Json json() {
        return new Json(getMoshi());
    }
}

File: templates/service-kotlin/{{#server.micronaut}}/src/main/kotlin/{{package.value}}/{{#jsonlib.moshi}}/MoshiConfig.kt
package {{package.value}};

import com.squareup.moshi.Moshi
import io.micronaut.context.annotation.*
import {{package.value}}.json.*

@Factory
class MoshiConfig {
    @Bean
    fun getMoshi(): Moshi {
        val moshiBuilder = Moshi.Builder()
        setupMoshiAdapters(moshiBuilder)
        return moshiBuilder.build()
    }

    @Bean
    fun json(): Json {
        return Json(getMoshi())
    }
}